Fix grammar (#2078)
flyteorg/flyte: rsts/concepts/versioning.rst

This may happen when different data scientists simultaneously iterate on the same workflow/pipeline.

The cost of creating an independent infrastructure for each version is enormous and not desirable.

> On the other hand, it is desirable to share the same centralized infrastructure, where the burden of maintaining the infrastructure is with a central infrastructure team,

Reviewer: ```suggestion
Moreover, it is desirable to share the same centralized infrastructure, where the burden of maintaining the infrastructure is with a central infrastructure team,
```
